plugin/core/views.py

```diff
@@ -100,15 +100,26 @@ def render_text_change(change: sublime.TextChange) -> Dict[str, Any]:
 
 
 def did_change_text_document_params(view: sublime.View,
-                                    changes: Optional[Iterable[sublime.TextChange]] = None) -> Dict[str, Any]:
+                                    changes: Optional[Sequence[sublime.TextChange]] = None) -> Dict[str, Any]:
     content_changes = []  # type: List[Dict[str, Any]]
     result = {"textDocument": versioned_text_document_identifier(view), "contentChanges": content_changes}
     if changes is None:
         # TextDocumentSyncKindFull
         content_changes.append({"text": entire_content(view)})
-    else:
+    elif len(changes) > 0:
         # TextDocumentSyncKindIncremental
-        for change in changes:
+        stitched_changes = []
+        current = changes[0]
+        for i in range(1, len(changes)):
+            candidate = changes[i]
+            length = len(current.str)
+            if current.b.pt + length == candidate.a.pt and candidate.a.pt == candidate.b.pt:
```

Comment:
I guess doing the latter check would be a bit less expensive.
And also, the `len(current.str)` expression could be inlined in the `if` so that we don't compute length unless `candidate.a.pt == candidate.b.pt`.